inc/saiswitch.h

```diff
@@ -1113,98 +1236,73 @@ typedef enum _sai_switch_attr_t
-    sai_port_event_notification_fn          on_port_event;
-    sai_switch_shutdown_request_fn          on_switch_shutdown_request;
-    sai_packet_event_notification_fn        on_packet_event;
-
-} sai_switch_notification_t;
-
-/**
- * @brief SDK initialization. After the call the capability attributes should
- * be ready for retrieval via sai_get_switch_attribute().
+ * @brief Create switch
  *
- * @param[in] profile_id Handle for the switch profile.
- * @param[in] switch_hardware_id Switch hardware ID to open
- * @param[in] firmware_path_name Vendor specific path name of the firmware to load
- * @param[in] switch_notifications Switch notification table
+ *   SDK initialization/connect to SDK. After the call the capability attributes should be
+ *   ready for retrieval via sai_get_switch_attribute().
  *
- * @return #SAI_STATUS_SUCCESS on success Failure status code on error
- */
-typedef sai_status_t (*sai_initialize_switch_fn)(
-        _In_ sai_switch_profile_id_t profile_id,
-        _In_reads_z_(SAI_MAX_HARDWARE_ID_LEN) char *switch_hardware_id,
-        _In_reads_opt_z_(SAI_MAX_FIRMWARE_PATH_NAME_LEN) char *firmware_path_name,
-        _In_ sai_switch_notification_t *switch_notifications);
-
-/**
- * @brief Release all resources associated with currently opened switch
+ * @param[out] switch_id The Switch Object ID
+ * @param[in] attr_count number of attributes
+ * @param[in] attr_list Array of attributes
  *
- * @param[in] warm_restart_hint Hint that indicates controlled warm restart.
- * Since warm restart can be caused by crash (therefore there are no guarantees
- * for this call), this hint is really a performance optimization.
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
  */
-typedef void (*sai_shutdown_switch_fn)(
-        _In_ bool warm_restart_hint);
+typedef sai_status_t(*sai_create_switch_fn)(
+        _Out_ sai_object_id_t* switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
 
 /**
- * @brief SDK connect. This API connects library to the initialized SDK. After
- * the call the capability attributes should be ready for retrieval via
- * sai_get_switch_attribute().
+ * @brief Remove/disconnect Switch
+ *   Release all resources associated with currently opened switch
  *
- * @param[in] profile_id Handle for the switch profile.
- * @param[in] switch_hardware_id Switch hardware ID to open
- * @param[in] switch_notifications Switch notification table
+ * @param[in] switch_id The Switch id
  *
  * @return #SAI_STATUS_SUCCESS on success Failure status code on error
  */
-typedef sai_status_t (*sai_connect_switch_fn)(
-        _In_ sai_switch_profile_id_t profile_id,
-        _In_reads_z_(SAI_MAX_HARDWARE_ID_LEN) char *switch_hardware_id,
-        _In_ sai_switch_notification_t *switch_notifications);
-
-/**
- * @brief Disconnect this SAI library from the SDK.
- */
-typedef void (*sai_disconnect_switch_fn)(void);
```

Comment:
how can we now disconnect from a switch without removing the switch object ?
with the old API app 1 could init switch
App 2 could connect and then disconnect, to free connection resources
With new API, app 2 can connect, but i don't see a way for it to disconnect